R/doeFactorial.R

```diff
@@ -175,100 +181,200 @@ doeFactorial <- function(jaspResults, dataset, options, ...) {
-                                          randomize = rnd))
-    } #blocks YES
+      rows <- FrF2::FrF2(nfactors = options[["numberOfFactors"]],
+                         nruns = 2^options[["numberOfFactors"]],
+                         ncenter = options[["factorialCenterPoints"]],
+                         replications = options[["factorialCornerReplicates"]],
+                         repeat.only = rep,
+                         blocks = options[["factorialBlocks"]],
+                         randomize = rnd)
+      runOrder <- 1:nrow(rows)
+      rows <- cbind.data.frame(runOrder, rows)
+    }
+
+    #p-b
+    if(options[["factorialType"]] == "factorialPlackettBurman"){
+      rows <- FrF2::pb(nruns = as.numeric(options[["PBruns"]]),
+                       nfactors = options[["numberOfFactors"]],
+                       ncenter = options[["factorialCenterPoints"]],
+                       replications = options[["factorialCornerReplicates"]],
+                       repeat.only = rep,
+                       randomize = rnd)
+      runOrder <- 1:nrow(rows)
+      rows <- cbind.data.frame(runOrder, rows)
+    }
 
+    blocks <- rows$Blocks
+    rows <- rows[,!names(rows) %in% "Blocks"]
+
+    #filling in table
     for (i in 1:as.numeric(options[["numberOfFactors"]])) {
       colnames(rows)[i+1] <- factorNames[i]
       table$addColumnInfo(name = factorNames[i], title = factorNames[i], type = 'string')
     }
 
-    jaspResults[["displayDesign"]] <- table
+    if(options[["factorialBlocks"]] > 1 || options[["factorialCornerReplicates"]] > 1){
+      blocks <- rows$Blocks
+      rows <- rows[,!names(rows) %in% "Blocks"]
+      # table$addColumnInfo(name = 'blocks', title = gettext("Blocks"), type = 'string')
+
+      if(options[["factorialBlocks"]] == 1 && options[["factorialCornerReplicates"]] > 1 && options[["factorialCenterPoints"]] > 0){
+        for(i in 1:options[["factorialCornerReplicates"]]){
+          for(j in 1:options[["factorialCenterPoints"]]){
+            blocks[is.na(blocks)][1] <- as.character(i)
+          }
+        }
+      }
 
-    if (options[["runOrder"]] == "runOrderStandard") {
-      rows <- rows[-1]
-      rows <- rows[do.call(order, rows),]
-      # rows <- rows[rev(rownames(rows)),]
-      rows <- cbind(runOrder = 1:nrow(rows), rows)
+      # rows[ncol(rows) + 1] <- blocks
     }
 
-    if (options[["dataCoding"]] == "dataUncoded") {
-      table$setData(rows)
+    rows <- rows[,!names(rows) %in% "Blocks"]
+
+    #run order randomization REPLACE
+    # if (options[["runOrder"]] == "runOrderStandard") {
+    #   rows <- rows[-1]
+    #   rows <- rows[do.call(order, rows),]
+    #   # rows <- rows[rev(rownames(rows)),]
+    #   rows <- cbind(runOrder = 1:nrow(rows), rows)
+    # }
+
+    #DATA CODING
+    if(options[["factorialCenterPoints"]] >= 1){
+      rows[,2:(options[["numberOfFactors"]]+1)] <- sapply(rows[,2:(options[["numberOfFactors"]]+1)], as.numeric)
```

Comment:
```suggestion
      rows[,2:(options[["numberOfFactors"]]+1)] <- as.numeric(rows[,2:(options[["numberOfFactors"]]+1)])
```
unless rows is a matrix of lists I guess? same for line 316.

Reply:
✅